include/picongpu/particles/synchrotron/AlgorithmSynchrotron.hpp

```diff
@@ -0,0 +1,431 @@
+                    float_64 oneMinusDeltaOverDelta = (1 - delta) / delta;
+                    float_X zq = 2. / (3 * chi) / oneMinusDeltaOverDelta;
+
+                    // zq convert to index and F1 and F2
+                    const float_X zqExponent = math::log2(zq);
+                    const int16_t index
+                        = static_cast<int16_t>((zqExponent - minZqExponent) / stepWidthLogatihmicScale);
+
+                    float_X F1 = 0;
+                    float_X F2 = 0;
+                    float_X Ftemp = 0;
+
+                    if(index >= 0 && index < interpolationPoints - 2)
+                    {
+                        float_X zq1 = math::pow(2, minZqExponent + stepWidthLogatihmicScale * (index));
+                        float_X zq2 = math::pow(2, minZqExponent + stepWidthLogatihmicScale * (index + 1));
+                        float_X f = (zq - zq1) / (zq2 - zq1);
+
+                        /// @todo Test Logarithmic interpolation:
+                        //      1) how slow is it?
+                        //      2) how much more accurate it is? (propably not much)
+                        //      3) if not worth to do it replace with:
+                        //          F1 = F1F2DeviceBuff(DataSpace<2>{index, 0});
+                        //          F2 = F1F2DeviceBuff(DataSpace<2>{index, 1});
+
+                        // logInterpolate F1 and F2 -> passed by reference
+                        logInterpolation(index, F1, Ftemp, f, 0, F1F2DeviceBuff);
+                        logInterpolation(index, F2, Ftemp, f, 1, F1F2DeviceBuff);
+                    }
+
+                    // Calculating the numeric factor: dt * (e**2 * m_e * c /( hbar**2 * eps0 * 4 * np.pi))
+                    float_X numericFactor = DELTA_T
+                        * (ELECTRON_CHARGE * ELECTRON_CHARGE * ELECTRON_MASS * SPEED_OF_LIGHT
+                           / (HBAR * HBAR * EPS0 * 4 * PI));
+
+                    if constexpr(params::supressRequirementWarning == false)
+                    {
+                        /// @todo change and use unified requirement
+                        float_X requirement1 = numericFactor * 1.5 * math::pow(chi, 2. / 3.) / gamma;
+                        float_X requirement2 = numericFactor * 0.5 * chi / gamma;
+
+                        /// this warning means that the propability of generating a photon is high for given dt (higher
+                        /// than 10%) this means that we possibly generate photons every timestep and
+                        /// underestimate the radiation. The solution would be to implement substepping
+                        if(requirement1 > 0.1
+                           || requirement2 > 0.1) // if requirement is not met send warning at every iteration
+                            alpaka::atomicExch(worker.getAcc(), &(failedRequirementQBuff(DataSpace<1>{0})), 1);
+                    }
+
+                    numericFactor *= math::sqrt(3) / (2 * PI);
```

Comment:
**BUG** `math::sqrt(3)` will be integer sqrt.

Please use `pmacc::math::Pi<float_X>::doubleValue` see https://github.com/ComputationalRadiationPhysics/picongpu/blob/f3a169c41480590ef0e47f3b4f54d198d35456cb/include/pmacc/algorithms/math/defines/pi.hpp#L33-L40 to have the correct precision.

Reply:
done